melodies_monet/util/satellite_utilities.py

```diff
@@ -32,6 +32,56 @@ def vertical_regrid(input_press, input_values, output_press):
             out_array[y,x,:] = f(xnew)
     return out_array
 
+def mod_to_overpasstime(modobj,opass_tms):
+    '''
+    Interpolate model to satellite overpass time.
+
+    Parameters
+    ----------
+    modobj : xarray, model data
+    opass_tms : datetime64, satellite overpass local time
+
+    Output
+    ------
+    outmod : revised model data at local overpass time
+    '''
+    import pandas as pd
+    import xarray as xr
+    
+    nst, = opass_tms.shape
+    nmt, = modobj.time.shape
+    ny,nx = modobj.longitude.shape
+    
+    # Determine local time offset
+    local_utc_offset = np.zeros([ny,nx],dtype='timedelta64[ns]')
+    # pandas timedelta calculation doesn't work on ndarrays
+    for xi in np.arange(nx):
+        local_utc_offset[:,xi] = pd.to_timedelta((modobj['longitude'].isel(x=xi)/15).astype(np.int64),unit='h')
+    
+    # initialize local time as variable
+    modobj['localtime'] = (['time','y','x'],np.zeros([nmt,ny,nx],dtype='datetime64[ns]'))
+    # fill
+    for ti in np.arange(nmt):
+        modobj['localtime'][ti] = modobj['time'][ti].data + local_utc_offset
```

Comment:
You may want to round `lon/15` instead of truncating by converting to int. You could try the approach here to get around the `pd.to_timedelta` limitation:
```python
local_utc_offset = (modobj['longitude'] / 15).round().astype("timedelta64[h]")
```
Then you should be able to do
```python
modobj['localtime'] = modobj['time'] + local_utc_offset
```